tests/tt_metal/tt_metal/perf_microbenchmark/11_remote_cb_sync_matmul_single_core/kernels/in0_reader.cpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: © 2024Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "dataflow_api.h"
+#include "ttnn/cpp/ttnn/operations/ccl/kernel_common/worker_sync_utils.hpp"
```

Comment:
I don't think we should be pulling from ttnn here for a metal test. Could you just copy the fn you need from it? Could put it in a file in the microbenchmark common folder if needed for multiple kernels and include from there